### PR TITLE
22 linalg ops

### PR DESCRIPTION
**Pull request by mlourakis**
Optimizations addressing #22, minor other revisions.

### CODE REVIEW

File: sqpnp/sqpnp.cpp [hunk 1]
```diff
@@ -1,13 +1,15 @@
...
-// George Terzakis (terzakig-at-hotmail-dot-com), September 2020
 // 
 // Implementation of SQPnP as described in the paper:
 //
 // "A Consistently Fast and Globally Optimal Solution to the Perspective-n-Point Problem" by G. Terzakis and M. Lourakis
 //     a) Paper:         https://www.ecva.net/papers/eccv_2020/papers_ECCV/papers/123460460.pdf
 //     b) Supplementary: https://www.ecva.net/papers/eccv_2020/papers_ECCV/papers/123460460-supp.pdf
+//
+// George Terzakis (terzakig-at-hotmail-dot-com), September 2020
+// Optimizations by Manolis Lourakis, February 2022, February 2024
```
Comment (terzakig) [inline sqpnp/sqpnp.cpp:11]: Just have both names in the comment, no need to specify what you did, unless you want to.

File: sqpnp/sqpnp.cpp [hunk 2]
```diff
@@ -289,13 +342,12 @@ namespace sqpnp
     H(3, 4) = r[6] - dot_j5q2*H(3, 1) - dot_j5q4*H(3, 3); H(4, 4) = r[7] - dot_j5q2*H(4, 1) - dot_j5q4*H(4, 3); H(5, 4) = r[8] - dot_j5q2*H(5, 1) - dot_j5q4*H(5, 3);
     H(6, 4) = r[3] - dot_j5q3*H(6, 2); H(7, 4) = r[4] - dot_j5q3*H(7, 2); H(8, 4) = r[5] - dot_j5q3*H(8, 2);
     
-    H.block<9, 1>(0, 4) /= H.col(4).norm();
+    H.block<9, 1>(0, 4) *= (1.0 / H.col(4).norm());
```
Comment (terzakig) [inline sqpnp/sqpnp.cpp:345]: No need for parenthesis.

File: sqpnp/sqpnp.cpp [hunk 3]
```diff
@@ -314,7 +366,7 @@ namespace sqpnp
     H(7, 5) = r[1] - dot_j6q3*H(7, 2) - dot_j6q5*H(7, 4); 
     H(8, 5) = r[2] - dot_j6q3*H(8, 2) - dot_j6q5*H(8, 4);
     
-    H.block<9, 1>(0, 5) /= H.col(5).norm();
+    H.block<9, 1>(0, 5) *= (1.0 / H.col(5).norm());
```
Comment (terzakig) [inline sqpnp/sqpnp.cpp:369]: Same as above.

File: sqpnp/sqpnp.cpp [hunk 4]
```diff
@@ -368,15 +421,17 @@ namespace sqpnp
     }
     const auto& v2 = Pn.block<9, 1>(0, index2);
     N.block<9, 1>(0, 1) = v2 - v2.dot( N.col(0) ) * N.col(0);
-    N.block<9, 1>(0, 1) /= N.col(1).norm();
+    N.block<9, 1>(0, 1) *= (1.0 / N.col(1).norm());
```
Comment (terzakig) [inline sqpnp/sqpnp.cpp:424]: Same as above, no need for parenthesis.

File: sqpnp/sqpnp.cpp [hunk 5]
```diff
@@ -390,7 +445,7 @@ namespace sqpnp
     const auto& v3 = Pn.block<9, 1>(0, index3);
     
     N.block<9, 1>(0, 2) = v3 - ( v3.dot( N.col(1) ) * N.col(1) ) - ( v3.dot( N.col(0) ) * N.col(0) );
-    N.block<9, 1>(0, 2) /= N.col(2).norm();
+    N.block<9, 1>(0, 2) *= (1.0 / N.col(2).norm());
```
Comment (terzakig) [inline sqpnp/sqpnp.cpp:448]: Same?

**Review by terzakig — CHANGES_REQUESTED** [review]
I assume `AxbSolveLDLt3x3` is tested on its own. Test runs OK, some minor comments.

File: sqpnp/sqpnp.cpp (hunk 1 above)
Comment (mlourakis) [reply sqpnp/sqpnp.cpp:11]: It's reminder for me as the optimizations were done in two batches...

File: sqpnp/sqpnp.cpp (hunk 2 above)
Comment (mlourakis) [reply sqpnp/sqpnp.cpp:347]: True but I find the code more readable with them.

**Pull request merged**